conjure/objectwriter.go

```diff
@@ -28,10 +29,28 @@ const (
 	dataVarName     = "data"
 )
 
+// To avoid re-computing safety recursively, we keep the computed result in this global cache.
+var safetyCache = make(map[types.Type]spec.LogSafety)
+
 func writeObjectType(file *jen.Group, objectDef *types.ObjectType) {
 	// Declare struct type with fields
 	containsCollection := false // If contains collection, we need JSON methods to initialize empty values.
-	file.Add(objectDef.Docs.CommentLine()).Type().Id(objectDef.Name).StructFunc(func(structDecl *jen.Group) {
+
+	// Compute the overall safety of the struct and add comment-based annotation
+	overallSafety := computeStructSafety(objectDef.Fields)
+	var safetyComment jen.Code = jen.Null()
+	if !overallSafety.IsUnknown() {
+		switch overallSafety.Value() {
+		case spec.LogSafety_SAFE:
+			safetyComment = jen.Comment("safelogging:@Safe")
+		case spec.LogSafety_UNSAFE:
+			safetyComment = jen.Comment("safelogging:@Unsafe")
+		case spec.LogSafety_DO_NOT_LOG:
+			safetyComment = jen.Comment("safelogging:@DoNotLog")
+		}
```

Comment:
Note that, since Go doesn't have sum types, if a new type is added here it will just be unhandled and `safetyCommnent` will be blank.

This could go either way, but given that this is within the context of a CLI/developer tool, I think it may make sense to have a `default` fallthrough case that panics